TestProject/SimpleTests.cs.expected

```diff
@@ -0,0 +1,202 @@
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using NUnit.Framework;
+using Rhino.Mocks;
+using TestProject.TestClasses;
+using TestProject.TestInterfaces;
+using Moq;
+using MockRepository = Rhino.Mocks.MockRepository;
+
+namespace TestProject
+{
+  [TestFixture]
+  public class SimpleTest
+  {
+    private Mock<ITestInterface> _mock1;
+    private Mock<ITestInterface> _mock2;
+
+    [SetUp]
+    public void Setup ()
+    {
+    }
+
+    [Test]
+    public void Rewrite_MockInstantiation ()
+    {
+      _ = new Mock<ITestInterface>();
+      _ = new Mock<ITestInterface>();
+      _ = new Mock<ITestInterface> (MockBehavior.Strict);
+      _ = new Mock<ITestInterface>() { CallBase = true };
+
+      _ = new Mock<ITestInterface>();
+      _ = new Mock<ITestInterface>();
+      _ = new Mock<ITestInterface> (MockBehavior.Strict);
+      _ = new Mock<ITestInterface> (MockBehavior.Strict);
+      _ = new Mock<ITestInterface>();
+      _ = new Mock<ITestInterface>() { CallBase = true };
+      _ = new Mock<ITestInterface>() { CallBase = true };
+
+      Assert.Pass();
+    }
+
+    [Test]
+    public void Rewrite_VerifyAll ()
+    {
+      _mock1 = new Mock<ITestInterface> (MockBehavior.Strict);
+      _mock2 = new Mock<ITestInterface>();
+
+      _mock1.Verify();
+      _mock2.Verify();
+
+      Assert.Pass();
+    }
+
+    [Test]
+    public void Rewrite_VerifyAllExpectations ()
+    {
+      _mock1.Verify();
+
+      Assert.Pass();
+    }
+
+    [Test]
+    public void Rewrite_AssertWasCalled ()
+    {
+      _mock1.Verify (_ => _.Read(), Times.AtLeastOnce());
+
+      Assert.Pass();
+    }
+
+    [Test]
+    public void Rewrite_AssertWasNotCalled ()
+    {
+      _mock1.Verify (_ => _.Read(), Times.Never());
+
+      Assert.Pass();
+    }
+
+    [Test]
+    public void Rewrite_RepeatAny ()
+    {
+      _mock1.Setup (m => m.Read (null)).Verifiable();
+
+      _mock1.Verify();
+
+      Assert.Pass();
+    }
+
+    [Test]
+    public void Rewrite_RepeatOnce ()
+    {
+      _mock1.Setup (m => m.Read (null)).Returns (true).Callback (null).Verifiable();
+
+      _mock1.Verify (m => m.Read (null), Times.Once());
+
+      Assert.Pass();
+    }
```

Comment:
```suggestion
    public void Rewrite_RepeatOnce ()
    {
      _mock1.Setup (m => m.Read (null)).Returns (true).Verifiable();
      
      _sut.Do();

      _mock1.Verify (m => m.Read (null), Times.Once());

      Assert.Pass();
    }
```

Reply:
We should probably demonstrate that the verify-statement is added in the correct place. Given that the verify-statement is more advanved than your typical replacement of `_mock1.VerifyAllExpectations();` it may be helpful to document that we didn't just add the new verify as a second statement after the first. 

In that line of thought, I think we need a version that uses stubs and no verify for the RhinoMocks syntax, too. With stubs, you can also also play with number-of-repeats (I think) and when there's more calls, RhinoMocks falls back to default behavior. Either way, there's no verify-statment and that might result in problems. I cannot say right now, if this scenario actually exists in our code base, but we should have an anwser to a RhinoMocks repeat without a verify.

Same also goes for strict-mocks without a verify. That's perfectly legal if you only want to make sure there's no extra calls but don't care to verify via RhinoMocks that the expected calls happened.